src/goto-instrument/contracts/dynamic-frames/dfcc_contract_clauses_codegen.cpp

```diff
@@ -116,6 +118,8 @@ void dfcc_contract_clauses_codegent::encode_assignable_target_group(
 
   auto label_instruction = dest.add(goto_programt::make_skip(source_location));
   goto_instruction->complete_goto(label_instruction);
+
+  destruct_locals(new_vars, dest, ns);
```

Comment:
What is the general rule for when do kill these temporaries in the presence of jumps ?

Reply:
Added as documentation: they need to be killed on all control-flow paths.